consensus/spos/sposFactory/sposFactory.go

```diff
@@ -77,6 +78,17 @@ func GetBroadcastMessenger(
 		return nil, spos.ErrNilShardCoordinator
 	}
 
+	dbbArgs := &broadcast.ArgsDelayedBlockBroadcaster{
+		InterceptorsContainer: interceptorsContainer,
+		HeadersSubscriber:     headersSubscriber,
+		ShardCoordinator:      shardCoordinator,
+		LeaderCacheSize:       maxDelayCacheSize,
+		ValidatorCacheSize:    maxDelayCacheSize,
+		AlarmScheduler:        alarmScheduler,
+	}
+
+	delayedBroadcaster, _ := broadcast.NewDelayedBlockBroadcaster(dbbArgs)
```

Comment:
here we should check the returned error and propagate it in the result of the method. Otherwise we may get a nil delayedBroadcaster here, which may lead to panics.

```
if err != nil {
   return nil, err
}
```

Reply:
updated